tsdb/engine/tsm1/cache.go

```diff
@@ -180,37 +187,79 @@ func (c *Cache) Keys() []string {
 
 // Values returns a copy of all values, deduped and sorted, for the given key.
 func (c *Cache) Values(key string) Values {
```

Comment:
FWIW, if we have this function return `[]Values` we might be able to avoid copying the immutable snapshot slices. Since they are read-only, it's a shame to copy them. This call would instead promise to return a slice, ordered so that each `Values` element should be walked before the next one.

How much data do we think a snapshot could hold?